dpsim-models/src/DP/DP_Ph1_Transformer_deprecated.cpp

```diff
@@ -0,0 +1,304 @@
+  mSubSnubCapacitor2->setParameters(mSnubberCapacitance2);
+  mSubSnubCapacitor2->connect({node(1), DP::SimNode::GND});
+  SPDLOG_LOGGER_INFO(
+      mSLog,
+      "Snubber Capacitance 2 (connected to lower voltage side {}) = {} [F]",
+      node(1)->name(), Logger::realToString(mSnubberCapacitance2));
+  addMNASubComponent(mSubSnubCapacitor2,
+                     MNA_SUBCOMP_TASK_ORDER::TASK_BEFORE_PARENT,
+                     MNA_SUBCOMP_TASK_ORDER::TASK_BEFORE_PARENT, true);
+
+  // Initialize electrical subcomponents
+  SPDLOG_LOGGER_INFO(mSLog, "Electrical subcomponents: ");
+  for (auto subcomp : mSubComponents) {
+    SPDLOG_LOGGER_INFO(mSLog, "- {}", subcomp->name());
+    subcomp->initialize(mFrequencies);
+    subcomp->initializeFromNodesAndTerminals(frequency);
+  }
+
+  SPDLOG_LOGGER_INFO(
+      mSLog,
+      "\n--- Initialization from powerflow ---"
+      "\nVoltage across: {:s}"
+      "\nCurrent: {:s}"
+      "\nTerminal 0 voltage: {:s}"
+      "\nTerminal 1 voltage: {:s}"
+      "\nVirtual Node 1 voltage: {:s}"
+      "\n--- Initialization from powerflow finished ---",
+      Logger::phasorToString((**mIntfVoltage)(0, 0)),
+      Logger::phasorToString((**mIntfCurrent)(0, 0)),
+      Logger::phasorToString(initialSingleVoltage(0)),
+      Logger::phasorToString(initialSingleVoltage(1)),
+      Logger::phasorToString(mVirtualNodes[0]->initialSingleVoltage()));
+}
+
+void DP::Ph1::Transformer::mnaParentInitialize(
+    Real omega, Real timeStep, Attribute<Matrix>::Ptr leftVector) {
+  SPDLOG_LOGGER_INFO(
+      mSLog,
+      "\nTerminal 0 connected to {:s} = sim node {:d}"
+      "\nTerminal 1 connected to {:s} = sim node {:d}",
+      mTerminals[0]->node()->name(), mTerminals[0]->node()->matrixNodeIndex(),
+      mTerminals[1]->node()->name(), mTerminals[1]->node()->matrixNodeIndex());
+}
+
+void DP::Ph1::Transformer::mnaCompApplySystemMatrixStamp(
+    SparseMatrixRow &systemMatrix) {
+  // Ideal transformer equations
+  if (terminalNotGrounded(0)) {
+    Math::setMatrixElement(systemMatrix, mVirtualNodes[0]->matrixNodeIndex(),
+                           mVirtualNodes[1]->matrixNodeIndex(),
+                           Complex(-1.0, 0));
+    Math::setMatrixElement(systemMatrix, mVirtualNodes[1]->matrixNodeIndex(),
+                           mVirtualNodes[0]->matrixNodeIndex(),
+                           Complex(1.0, 0));
+  }
+  if (terminalNotGrounded(1)) {
+    Math::setMatrixElement(systemMatrix, matrixNodeIndex(1),
+                           mVirtualNodes[1]->matrixNodeIndex(), **mRatio);
+    Math::setMatrixElement(systemMatrix, mVirtualNodes[1]->matrixNodeIndex(),
+                           matrixNodeIndex(1), -**mRatio);
+  }
+
+  // Add subcomps to system matrix
+  for (auto subcomp : mSubComponents)
+    if (auto mnasubcomp = std::dynamic_pointer_cast<MNAInterface>(subcomp))
+      mnasubcomp->mnaApplySystemMatrixStamp(systemMatrix);
+
+  if (terminalNotGrounded(0)) {
```

Comment:
I think we can log this in the same place that the code is (with the equations) on line 221